baseband/vlbi_base/tests/test_vlbi_base.py

```diff
@@ -426,142 +427,148 @@ def test_vlbi_file_base(self, tmpdir):
+         (True, (0, 13), (2,), ('n4',)),
+         (False, (Ellipsis, 0, 1), (1, 21, 33), None),
+         (True, (Ellipsis, 0, 1), (21,), ('n1',)),
+         (False, (0, slice(1, None, 4), slice(None), 0),
+          (5, 33, 2), ('n1', 'n2', 'n4')),
+         (True, (slice(1, None, 4), slice(None), [1]),
+          (5, 33, 1), ('n1', 'n2', 'n4')),
+         (False, (0, 0, slice(None, 1, -4)), (8, 1, 2), ('n2', 'n3', 'n4')),
+         (True, (0, slice(None, 1, -4)), (8, 2), ('n2', 'n4')),
+         (False, (0, np.array([2, 8, 9])[:, np.newaxis], [1, 7]),
+          (3, 2, 1, 2), None),
+         (True, (np.array([2, 8, 9])[:, np.newaxis], [1, 7], 0),
+          (3, 2), None)])
+    def test_squeeze_subset(self, squeeze, subset, sliced_shape, sliced_n):
+        # Tests subsetting for squeezed samples.
+        sb = self.make_reader_with_shape(squeeze=squeeze, subset=subset)
+        if isinstance(subset, tuple):
+            assert sb.subset == subset
+        elif subset is None:
+            assert sb.subset == ()
+        else:
+            assert sb.subset == (subset,)
         assert sb.sample_shape == sliced_shape
-        subset_data = data[(slice(0, 12),) + sb.subset]
-        assert subset_data.shape == (12,) + sb.sample_shape
-
-        # Check that squeezing subset data is self-consistent and consistent
-        # with squeezing the data from above.
-        sbs = self.make_writer_with_shape(sample_shape_maker, unsliced_shape,
-                                          subset=subset, squeeze=True)
-        assert sbs.sample_shape == squeezed_shape
-        assert sbs.sample_shape._fields == squeezed_n
-        sbs_data = data[(slice(0, 12),) + sbs.subset]
-        assert sbs_data.squeeze().shape == (12,) + sbs.sample_shape
-        assert sbs_data.squeeze().shape == subset_data.squeeze().shape
-
-    def test_subset_specialindexing(self):
-        # Tests that single integer subsets are converted to slices if
-        # squeeze is False, but any advanced indexing that reduces
-        # dimensionality returns IndexErrors.
-
-        sample_shape_maker = namedtuple('SampleShape',
-                                        'n0, n1, n2, n3, n4')
-        unsliced_shape = (1, 21, 33, 4, 1)
-
-        # When squeeze = False, passing ints returns slices.
-        sb = self.make_writer_with_shape(sample_shape_maker, unsliced_shape,
-                                         subset=(0, -1), squeeze=False)
-        assert sb.subset == (slice(0, 1), slice(-1, None))
-
-        # When squeeze = True, passing ints returns ints.
-        sb = self.make_writer_with_shape(sample_shape_maker, unsliced_shape,
-                                         subset=(0, -1), squeeze=True)
-        assert sb.subset == (0, -1)
+        assert getattr(sb.sample_shape, '_fields', None) == sliced_n
+        subset_data = sb._squeeze_and_subset(self.unsliced_data)
+        assert subset_data.shape == (100,) + sliced_shape
 
+    def test_failty_subset(self):
```

Comment:
Should this be `test_faulty_subset`?  (Though "failty" does exist on UrbanDictionary, so not certain...)